ucsb_ds_capstone_projects_2021/projects/carpedata/update1.md

```diff
@@ -1,15 +1,52 @@
+
+Given our large dataset, we started by examining a random sample. Using pandas, we randomly selected 5% of the dataset. From here we were able to prototype a function to parse through the html, break the sentence up into individual words, and then remove all punctuation, case, and stopwords (words which convey not meaningful information). This data we fed into a TFIDFvectorizer – TFIDF means term frequency - inverse document frequency. In short, the TFIDF vectorizer turns each document into a numerical feature, weighting the words by their frequency across the entire set of documents. This decreases the importance of very common and meaningless words while excluding words that occur above or below our desired frequency all together. With our new vectorized features we were able to train a logistic regression model to promising results: 94% mean accuracy, 87% recall and 89% precision.  
+Seeing the early success of our parsing function and a fairly simplistic model, we expanded our methods to the entire dataset. Our basic parsing/cleaning function was beefed up into a script that could run overnight. Parsing the html, tokenizing the strings, and then removing unnecessary attributes is extremely computationally expensive; our training data set took approximately 15 hours to be cleaned. In the end our training dataset was reduced from 4Gb to 628 MB and our test dataset was reduced from 1Gb to 151 MB. Training a logistic regression model on the full dataset again yielded excellent results - 94% accuracy, 94% recall, and 85% precision. 
+
+#### Sampling our Data
+
+<img width="744" alt="Screen Shot 2021-03-15 at 5 25 54 PM" src="https://user-images.githubusercontent.com/43657731/111238047-8449c780-85b3-11eb-8620-204522bf8594.png">
+
+#### Cleaning Function
+
+```
+def clean_and_tokenize(x):
+    """Takes in string of html codes and returns cleaned and tokenized 
+    requires BeautifulSoup, nltk.word_tokenize, nltk.corpus stopwords"""
+    #stemmer = PorterStemmer()
+    soup = BeautifulSoup(x, 'lxml') #parses html
+    text = soup.get_text() #convert to text
+    tokens = word_tokenize(text) #tokenize
+    stop_words = set(stopwords.words('english')) #remove stop words
+    # remove all tokens that are not alphabetic 
+    words = [word.lower() for word in tokens if word.isalpha() and word not in stop_words]
+    w = " ".join(words)
+    return w
+  ```
+  
+### Logistic Regression 
+ 
+ The classification report of our logistic regression showed a promissing model. 
+ 
+<img width="483" alt="Screen Shot 2021-03-15 at 5 29 31 PM" src="https://user-images.githubusercontent.com/43657731/111238284-04702d00-85b4-11eb-9027-c6fa7d20a767.png">
+
+Examing the words associated with the largest coefficients of our model showed some seemingly intuitive results.
+ 
+<img width="672" alt="Screen Shot 2021-03-15 at 5 30 22 PM" src="https://user-images.githubusercontent.com/43657731/111238348-22d62880-85b4-11eb-83c4-1dd7e1fbf8da.png">
```

Comment:
Same here
Markdown can embed images directly: https://jupyterbook.org/content/figures.html
For the image `someimg.png` in the same directory as the markdown file, replace the HTML with `![some-text](someimg.png)`